lib/as2.js

```diff
@@ -0,0 +1,600 @@
+                    "favorites": "liked"
+                };
+                _.each(streams, function(bfunc, stream) {
+                    if (_.isString(obj[stream])) {
+                        builder[bfunc](obj[stream]);
+                    } else if (_.isObject(obj[stream]) && _.isString(obj[stream].url)) {
+                        builder[bfunc](obj[stream].url);
+                    }
+                });
+                this(null, builder);
+            },
+            cb
+        );
+    }
+};
+
+// Special case map for objectType -> as method name
+var objectVocabMap = {
+
+};
+
+// returns the name of the best as method to use for creating an as2 object
+// that approximates this type
+
+var objectTypeToMethod = function(objectType) {
+    if (!objectType) {
+        return "object";
+    } else if (_.has(objectVocabMap, objectType)) {
+        return objectVocabMap[objectType];
+    } else if (_.includes(ActivityObject.objectTypes, objectType) &&
+                _.has(as, objectType)) {
+        return objectType;
+    } else {
+        return "object";
+    }
+};
+
+// Converts an AS1 activity to an AS2 activity based on the verb
+
+var defaultActivityConverter = function(act, cb) {
+    Step(
+      function() {
+        activityBuilder(act, this);
+      },
+      function(err, builder) {
+        if (err) throw err;
+        // Use the builder to set properties
+        if (_.isFunction(activityProperties[act.verb])) {
+            activityProperties[act.verb](builder, act, this);
+        } else {
+            defaultActivityProperties(builder, act, this);
+        }
+      },
+      function(err, builder) {
+        if (err) throw err;
+        builder.get().export(this);
+      },
+      cb
+    );
+};
+
+// Returns an as builder based on the as1 activity
+
+var activityBuilder = function(act, cb) {
+  if (!act.verb) {
+    defaultActivityBuilder(act, cb);
+  } else if (_.has(activityBuilders, act.verb)) {
+    activityBuilders[act.verb](act, cb);
+  } else {
+    defaultActivityBuilder(act, cb);
+  }
+};
+
+// The default method for getting an as builder, but see below for special cases
+
+var defaultActivityBuilder = function(act, cb) {
+  assert(_.isObject(act));
+  var method = verbToMethod(act.verb);
+  assert(_.isFunction(as[method]));
+  var builder = as[method]();
+  cb(null, builder);
+};
+
+// Use the builder to set activity properties
+
+var defaultActivityProperties = function(builder, act, cb) {
+    // For ActivityPub, IDs should resolve to the thing
+    var self = _.get(act, "links.self.href");
+    if (self && self !== act.id) {
+        builder.id(self);
+    } else if (_.has(act, "id")) {
+        builder.id(act.id);
+    }
+    if (_.has(act, "displayName")) {
+        builder.name(act.displayName);
+    } else if (_.has(act, "title")) {
+        builder.name(act.title);
+    }
+    if (_.has(act, "content")) {
+        builder.summary(act.content);
+    }
+    if (_.has(act, "published")) {
```

Comment:
I think you can put this code in other function because of looks like the same at line 119